src/thunderbots/software/ai/main.cpp

```diff
@@ -109,6 +110,12 @@ int main(int argc, char **argv)
             std::cout << msg << std::endl;
         }
         primitive_publisher.publish(primitive_array_message);
+        }
+        catch (const std::invalid_argument& e)
```

Comment:
Hmm, should we be catching specific exceptions here? Or all exceptions? I had thought the idea was to make sure that no matter what, we could carry on if a given iteration of AI threw an exception. @MathewMacDougall ?

Reply:
No, the AI shouldn't be 100% fault-tolerant. There still may be cases where something is so terribly wrong we want to abort.

Catching specific exceptions is a way of implementing the "severity" we talked about above, where we can allow some exceptions to be caught and recovered from and other to terminate the AI (or be caught at a higher level).

Reply:
Fair enough, I agree with this!